File: tests/adapters/python/goldens/comments/keep_all.py
"""Module docstring with detailed description.

This module demonstrates various comment styles and documentation
patterns for testing comment optimization policies.
"""

import os
import sys  # Standard library import
from typing import List, Optional  # TODO: Consider using generics


class CommentedClass:
    """Class with various comment types."""
    
    def __init__(self, name: str):
        """Initialize with name.
        
        Args:
            name: The name to set
            
        Returns:
            None
        """
        # This is a regular comment
        self.name = name
        self.data = []  # FIXME: Should use better data structure
    
    def public_method(self) -> str:
        """Get the name value.
        
        This method returns the current name. It's a simple getter
        that provides access to the internal name field. 
        Very straightforward implementation.
        """
        # Regular comment before return
        return self.name
    
    def _private_method(self):
        """Private method with documentation.
        
        This is an internal utility method.
        Should not be used externally.
        """
        # Implementation details
        # Multiple line comment
        # explaining the logic
        temp = self.name.upper()
        return temp  # Another trailing comment
    
    # Class-level comment
    def method_with_lots_of_comments(self):
        # Start of method
        value = "test"
        
        # First processing step
        # This handles the initial data
        processed = value.strip()
        
        # Second processing step - validation
        if not processed:
            # Handle empty case
            return None
            
        # Final step - return result
        # NOTE: Could be optimized
        return processed.lower()

def standalone_function():
    """Function with minimal documentation."""
    # Just a simple comment
    pass

# Module-level comment
def undocumented_function():
    # No docstring, just comments
    # This function doesn't have proper documentation
    x = 1  # Simple assignment
    y = 2  # Another assignment  
    return x + y  # Return sum

# TODO: Implement better error handling
def function_with_annotations():
    """Function with TODO and FIXME annotations.
    
    This function needs improvement.
    """
    # FIXME: This logic is flawed
    result = "placeholder"
    # WARNING: Not implemented properly
    return result

if __name__ == "__main__":
    # Main execution comment
    obj = CommentedClass("test")
    print(obj.public_method())

File: tests/adapters/cpp/goldens/comments/keep_all.cpp
/**
 * C++ module for testing comment optimization.
 *
 * This module contains various types of comments to test
 * different comment processing policies and edge cases.
 */

#include <iostream>
#include <string>
#include <vector>
#include <regex>

// Single-line comment at module level
constexpr const char* MODULE_VERSION = "1.0.0";  // TODO: Move to config file

/**
 * Structure with documentation comments.
 * This should be preserved when keeping documentation comments.
 */
struct User {
    int id;            // User identifier
    std::string name;  // FIXME: Should validate name format
    std::string email; // User's email address
    // Optional profile data
    void* profile;
};

struct Profile {
    std::string bio;
    std::string avatar;
};

class CommentedService {
private:
    void* config;      // Service configuration
    void* logger;      // Optional logger

public:
    /**
     * Class constructor with detailed documentation.
     *
     * Initializes the service with the provided configuration
     * and sets up the logging system if logger is provided.
     *
     * @param config Service configuration object
     * @param logger Logger instance (can be nullptr)
     */
    CommentedService(void* config, void* logger)
        : config(config), logger(logger) {
        // Initialize service
        initialize();

        // TODO: Add configuration validation
        // FIXME: Logger should be required, not optional
    }

    /**
     * Process user data with validation.
     *
     * This method performs comprehensive user data processing including
     * validation, transformation, and persistence operations. It handles
     * various edge cases and provides detailed error reporting.
     *
     * @param userData The user data to process
     * @return The processed user
     * @throws std::runtime_error when data is invalid
     */
    User processUser(const User& userData) {
        // Pre-processing validation
        if (userData.name.empty()) {
            throw std::runtime_error("User data is required");
        }

        /*
         * Multi-line comment explaining
         * the validation logic that follows.
         * This is important business logic.
         */
        bool isValid = true;
        if (userData.name.empty()) {
            // Log validation failure
            std::cerr << "Validation failed: name is required" << std::endl;
            isValid = false;
        }

        if (userData.email.empty()) {
            std::cerr << "Validation failed: email is required" << std::endl;
            isValid = false;
        }

        if (!isValid) {
            throw std::runtime_error("Validation failed");
        }

        // Transform data for storage
        User transformedData = transformUserData(userData);

        // Persist to database
        // NOTE: This could be optimized with batch operations
        User savedUser = saveUser(transformedData);

        return savedUser;  // Return the saved user
    }

private:
    User transformUserData(const User& userData) {
        /*
         * Data transformation logic.
         * Convert partial user data to complete user object
         * with all required fields populated.
         */
        User user;
        user.id = generateUserId();      // Generate unique ID
        user.name = userData.name;       // Clean up name
        user.email = userData.email;     // Normalize email
        user.profile = userData.profile; // Optional profile

        return user;
    }

    /**
     * Generate unique user ID.
     * @return Generated user ID
     */
    int generateUserId() {
        // Simple ID generation
        return rand() % 1000000;
    }

    // TODO: Implement proper persistence layer
    User saveUser(const User& user) {
        // Simulate database save
        // In real implementation, this would use a database

        // Log save operation
        std::cout << "Saving user: " << user.id << std::endl;

        // Simulate async operation
        // (sleep or similar in real code)

        return user;  // Return saved user
    }

    void initialize() {
        // Service initialization
        // This method sets up the service state

        // TODO: Add proper initialization logic
        // WARNING: This is a placeholder implementation
    }
};

/**
 * Utility function with comprehensive documentation.
 *
 * @param input The input string to process
 * @return Processed string result
 */
std::string processString(const std::string& input) {
    // Input validation
    if (input.empty()) {
        return "";  // Return empty string for invalid input
    }

    /* Process the string:
     * 1. Trim whitespace
     * 2. Convert to lowercase
     * 3. Remove special characters
     */
    std::string trimmed = input;
    std::string lowercase = trimmed;
    std::transform(lowercase.begin(), lowercase.end(), lowercase.begin(), ::tolower);

    std::string cleaned = std::regex_replace(lowercase, std::regex("[^a-z0-9\\s]"), "");

    return cleaned;  // Return processed string
}

// Module-level function without documentation
void undocumentedHelper() {
    // This function has no documentation comments
    // Only regular comments explaining implementation

    // Implementation details...
    const char* data = "helper data";

    // Process data
    std::cout << data << std::endl;  // Log the data
}

// Type definitions with comments
struct ValidationResult {
    bool isValid;                    // Whether validation passed
    std::vector<std::string> errors; // List of validation errors
};

struct ServiceConfig {
    // Configuration options
    int timeout;          // Request timeout in milliseconds
    int retries;          // Number of retry attempts
    std::string baseUrl;  // Base URL for API calls
};

/*
 * Export default configuration
 * This is used when no custom config is provided
 */
ServiceConfig DEFAULT_CONFIG = {
    5000,                       // 5 second timeout
    3,                          // 3 retry attempts
    "http://localhost:3000"     // Default base URL
};
